src/client/Microsoft.Identity.Client/AppConfig/AuthorityInfo.cs

```diff
@@ -145,34 +145,47 @@ internal class AuthorityInfo
         #region Builders
         internal static AuthorityInfo FromAuthorityUri(string authorityUri, bool validateAuthority)
         {
-            string canonicalUri = CanonicalizeAuthorityUri(authorityUri);
-            ValidateAuthorityUri(canonicalUri);
+            string canonicalAuthority = CanonicalizeAuthorityUri(authorityUri);
+
+            Uri canonicalAuthorityUri = ValidateAndCreateAuthorityUri(canonicalAuthority);
 
-            var authorityType = GetAuthorityType(canonicalUri);
+            canonicalAuthorityUri = TransformIfCiamAuthority(canonicalAuthorityUri);
+
+            var authorityType = GetAuthorityType(canonicalAuthorityUri);
 
             // Authority validation is only supported for AAD 
             if (authorityType == AuthorityType.B2C || authorityType == AuthorityType.Generic)
             {
                 validateAuthority = false;
             }
 
-            return new AuthorityInfo(authorityType, canonicalUri, validateAuthority);
+            return new AuthorityInfo(authorityType, canonicalAuthorityUri, validateAuthority);
         }
 
-        internal static AuthorityInfo FromAadAuthority(Uri cloudInstanceUri, Guid tenantId, bool validateAuthority)
+        private static Uri TransformIfCiamAuthority(Uri authorityUri)
+        {
+            if (isCiamAuthority(authorityUri))
+            {
+                return CiamAuthority.TransformAuthority(authorityUri);
+            }
+
+            return authorityUri;
+        }
+
+        internal static AuthorityInfo FromAadAuthority(string cloudInstanceUri, Guid tenantId, bool validateAuthority)
         {
 #pragma warning disable CA1305 // Specify IFormatProvider
-            return FromAuthorityUri(new Uri(cloudInstanceUri, tenantId.ToString("D")).AbsoluteUri, validateAuthority);
+            return FromAuthorityUri($"{cloudInstanceUri.Trim().TrimEnd('/')}/{tenantId.ToString("D")}", validateAuthority);
```

Comment:
This method is called by an obsolete api. probably not super impactful but I still changed it. Added a `.Trim().TrimEnd('/')` to avoid a double `/`

@bgavrilMS